streams/src/main/java/org/apache/kafka/streams/kstream/internals/suppress/KTableSuppressProcessor.java

```diff
@@ -16,34 +16,66 @@
  */
 package org.apache.kafka.streams.kstream.internals.suppress;
 
+import org.apache.kafka.common.header.Header;
 import org.apache.kafka.common.serialization.Serde;
+import org.apache.kafka.streams.KeyValue;
+import org.apache.kafka.streams.errors.TopologyException;
 import org.apache.kafka.streams.kstream.internals.Change;
+import org.apache.kafka.streams.kstream.internals.FullChangeSerde;
 import org.apache.kafka.streams.processor.Processor;
 import org.apache.kafka.streams.processor.ProcessorContext;
+import org.apache.kafka.streams.processor.PunctuationType;
 import org.apache.kafka.streams.processor.internals.InternalProcessorContext;
+import org.apache.kafka.streams.processor.internals.ProcessorNode;
+import org.apache.kafka.streams.processor.internals.ProcessorRecordContext;
 
 import java.time.Duration;
+import java.util.Iterator;
 
 import static java.util.Objects.requireNonNull;
 
 public class KTableSuppressProcessor<K, V> implements Processor<K, Change<V>> {
     private final SuppressedImpl<K> suppress;
+    private final long maxRecords;
+    private final long maxBytes;
+    private final Duration suppressDuration;
     private InternalProcessorContext internalProcessorContext;
+    private TimeOrderedKeyValueBuffer<K, ContextualRecord<Change<V>>> buffer;
 
-    private final Serde<K> keySerde;
-    private final Serde<Change<V>> valueSerde;
+    private ProcessorNode myNode;
+    private Serde<K> keySerde;
+    private Serde<Change<V>> valueSerde;
 
     public KTableSuppressProcessor(final SuppressedImpl<K> suppress,
                                    final Serde<K> keySerde,
                                    final Serde<Change<V>> valueSerde) {
         this.suppress = requireNonNull(suppress);
         this.keySerde = keySerde;
         this.valueSerde = valueSerde;
+        maxRecords = suppress.getBufferConfig().maxRecords();
+        maxBytes = suppress.getBufferConfig().maxBytes();
+        suppressDuration = suppress.getTimeToWaitForMoreEvents();
+    }
+
+    private void ensureBuffer() {
```

Comment:
As demonstrated by Part 1, we don't always need the buffer, so I thought it best to avoid allocating it and scheduling the associated punctuator until we first discover we need to buffer something.